spec/models/workflow/instance/process_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe Workflow::Instance::Process, type: :model do
+  let(:process) { create(:workflow_instance_process) }
+  describe "when the instance does not have its own categories" do
+    it "fetches categories from the definition" do
+      expect(process.category_list).to be_empty
+      process.definition.category_list.add("Finance")
+      process.definition.save!
+      expect(process.reload.category_list).to_not be_empty
+    end
+  end
+end
```

Comment:
Do you mind testing the positive case too?  When it does have its own categories, it uses that over definition?

Reply:
Good call. When i tried to test the positive case, i realized my "clever" method was providing a bit too much magic. There was no else statement, and more dangerously, you'd always be adding categories to the definition.  `#category_list.add` would return and add new categories to the definition's category list. 

I created a helper function, `get_categories` and put it into the concern.